src/test/unit/services/pathfinder/normal_glm_test.cpp

```diff
@@ -0,0 +1,297 @@
+  //       std::cout << "Mean Values: \n" <<
+  //       mean_vals.transpose().eval().format(CommaInitFmt) << "\n";
+  Eigen::VectorXd sd_vals = (((param_vals.colwise() - mean_vals)
+                                  .array()
+                                  .square()
+                                  .matrix()
+                                  .rowwise()
+                                  .sum()
+                                  .array()
+                              / (param_vals.cols() - 1))
+                                 .sqrt())
+                                .transpose()
+                                .eval();
+  // std::cout << "\n\n" << param_vals.format(CommaInitFmt) << "\n\n";
+  // std::cout << "rows: " << param_vals.rows() << " cols: " <<
+  // param_vals.cols() << "\n";
+  Eigen::MatrixXd prev_param_vals = stan::test::normal_glm_param_vals();
+  // std::cout << "\n\n" << prev_param_vals.format(CommaInitFmt) << "\n\n";
+  Eigen::VectorXd prev_mean_vals = prev_param_vals.rowwise().mean().eval();
+  //       std::cout << "Mean Values: \n" <<
+  //       mean_vals.transpose().eval().format(CommaInitFmt) << "\n";
+  Eigen::VectorXd prev_sd_vals = (((prev_param_vals.colwise() - prev_mean_vals)
+                                       .array()
+                                       .square()
+                                       .matrix()
+                                       .rowwise()
+                                       .sum()
+                                       .array()
+                                   / (prev_param_vals.cols() - 1))
+                                      .sqrt())
+                                     .transpose()
+                                     .eval();
+  Eigen::MatrixXd ans_diff = param_vals - prev_param_vals;
+  Eigen::VectorXd mean_diff_vals = ans_diff.rowwise().mean();
+  //      std::cout << "diff Mean Values: \n" <<
+  //      mean_diff_vals.transpose().eval().format(CommaInitFmt) << "\n";
+  Eigen::VectorXd sd_diff_vals = (((ans_diff.colwise() - mean_diff_vals)
+                                       .array()
+                                       .square()
+                                       .matrix()
+                                       .rowwise()
+                                       .sum()
+                                       .array()
+                                   / (ans_diff.cols() - 1))
+                                      .sqrt())
+                                     .transpose()
+                                     .eval();
+
+  Eigen::MatrixXd all_mean_vals(3, 10);
+  all_mean_vals.row(0) = mean_vals;
+  all_mean_vals.row(1) = prev_mean_vals;
+  all_mean_vals.row(2) = mean_diff_vals;
+
+  Eigen::MatrixXd all_sd_vals(3, 10);
+  all_sd_vals.row(0) = sd_vals;
+  all_sd_vals.row(1) = prev_sd_vals;
+  all_sd_vals.row(2) = sd_diff_vals;
+  for (int i = 0; i < all_mean_vals.cols() - 2; ++i) {
+    EXPECT_NEAR(0, all_mean_vals(2, i), 1);
```

Comment:
This is a very wide test with 1 as the tolerance.  And why just test means and not the other values?  Or I should say, why compute the other values if they're not being tested?  

Should we put in values from Lu's R code to test against here?

Reply:
> This is a very wide test with 1 as the tolerance. 
> And why just test means and not the other values? 
> Or I should say, why compute the other values if they're not being tested?

Yeah that tolerance can be lowered to .01 for means and .1 for standard deviations. I just added tests for the sd values as well

> Should we put in values from Lu's R code to test against here?

Yes sorry prev_vals is actually the output from Lu's R code. I'll update the name to reflect that